app/models/mixins/read_only_mixin.rb

```diff
@@ -0,0 +1,17 @@
+module ReadOnlyMixin
+  extend ActiveSupport::Concern
+
+  included do
+    before_destroy :reject_if_read_only
+  end
+
+  private
+
+  def reject_if_read_only
+    if read_only? && !EvmDatabase.seeding?
+      model_name = self.class.name.pluralize.underscore.tr('_', ' ').gsub('miq ', '')
+      errors.add(:base, _("Read only #{model_name} cannot be deleted."))
```

Comment:
This is not a valid i18n structure...should be:

```suggestion
      errors.add(:base, _("Read only %{model_name} cannot be deleted.") % model_name)
```

Reply:
Also I'm not sure if that reads correctly?  The model name itself should probably _also_ be translated.  I think you have to use the Dictionary for that.